inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -1149,3 +1149,27 @@ def load_record_from_hep(obj, wf):
     obj.extra_data['head_revision_id'] = record_data['revision_id']
 
     return obj
+
+
+def remove_inspire_categories_derived_from_core_arxiv_categories(obj, eng):
+    if not obj.data.get("arxiv_eprints"):
+        return
+    inspire_categories_without_arxiv_sourced = [
+        category
+        for category in obj.data.get("inspire_categories")
+        if category.get("source") != "arxiv"
```

Comment:
From the requirements, shouldn't be `=='arxiv'`?

> The easiest approach is probably to call classify_field again in the workflow and substitute all inspire_categories with source=arXiv with the newly derived ones.

Reply:
No, it's correct, later we process  arxiv categories and we reclassify them leaving non-arxiv sourced categories not touched.

Reply:
No, it's correct, later we process arxiv categories and we reclassify them leaving non-arxiv sourced categories not touched.